apps/ledger-live-mobile/src/screens/SyncOnboarding/SoftwareChecksStep.tsx

```diff
@@ -0,0 +1,402 @@
+      if (firmwareUpdateStatus === "completed") {
+        firmwareUpdateUiStepStatus = "completed";
+      } else if (firmwareUpdateStatus === "failed") {
+        firmwareUpdateUiStepStatus = "failed";
+      }
+    }
+  }
+
+  // Handles the genuine check UI step title
+  let genuineCheckStepTitle;
+  switch (genuineCheckUiStepStatus) {
+    case "active":
+      genuineCheckStepTitle = t(
+        "syncOnboarding.softwareChecksSteps.genuineCheckStep.active.title",
+        {
+          productName,
+        },
+      );
+      break;
+    case "completed":
+      genuineCheckStepTitle = t(
+        "syncOnboarding.softwareChecksSteps.genuineCheckStep.completed.title",
+        {
+          productName,
+        },
+      );
+      break;
+    case "failed":
+      genuineCheckStepTitle = t(
+        "syncOnboarding.softwareChecksSteps.genuineCheckStep.failed.title",
+      );
+      break;
+    default:
+      genuineCheckStepTitle = t(
+        "syncOnboarding.softwareChecksSteps.genuineCheckStep.inactive.title",
+      );
+      break;
+  }
+
+  // Handles the firmware update UI step title
+  let firmwareUpdateStepTitle;
+  switch (firmwareUpdateUiStepStatus) {
+    case "active":
+      firmwareUpdateStepTitle = t(
+        "syncOnboarding.softwareChecksSteps.firmwareUpdateStep.active.title",
+      );
+      break;
+    case "completed":
+      if (
+        latestFirmwareGettingStatus === "available-firmware" &&
+        latestFirmware
+      ) {
+        firmwareUpdateStepTitle = t(
+          "syncOnboarding.softwareChecksSteps.firmwareUpdateStep.completed.updateAvailable.title",
+          {
+            firmwareVersion: JSON.stringify(latestFirmware.final.name),
+          },
+        );
+      } else {
+        firmwareUpdateStepTitle = t(
+          "syncOnboarding.softwareChecksSteps.firmwareUpdateStep.completed.noUpdateAvailable.title",
+        );
+      }
+      break;
+    case "failed":
+      firmwareUpdateStepTitle = t(
+        "syncOnboarding.softwareChecksSteps.firmwareUpdateStep.failed.title",
+      );
+      break;
+    default:
+      firmwareUpdateStepTitle = t(
+        "syncOnboarding.softwareChecksSteps.firmwareUpdateStep.inactive.title",
+      );
+      break;
+  }
+
+  // If there is already a displayed drawer, the currentDisplayedDrawer would be
+  // synchronized with nextDrawerToDisplay during the displayed drawer onClose event.
+  // Otherwise, currentDisplayDrawer needs to be set to nextDrawerToDisplay manually
+  if (currentDisplayedDrawer === "none" && nextDrawerToDisplay !== "none") {
+    setCurrentDisplayedDrawer(nextDrawerToDisplay);
+  }
+
+  return (
+    <Flex>
+      {isDisplayed && (
+        <Flex>
+          <GenuineCheckDrawer
+            productName={productName}
+            isOpen={
+              currentDisplayedDrawer === "requested" &&
+              nextDrawerToDisplay === "requested"
+            }
+            onPress={() => setGenuineCheckStatus("ongoing")}
```

Comment:
Possible to use useCallbacks for all these callbacks in the rendering?

Reply:
`GenuineCheckDrawer` is not a memoized component, so it will re-render whenever its parent re-renders